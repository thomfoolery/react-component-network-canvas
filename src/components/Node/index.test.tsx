import React from "react";
import renderer from "react-test-renderer";

import { Node } from "./index";

describe("Node", () => {
  it("renders correctly", () => {
    const node = {
      id: "node-1",
      position: {
        x: 0,
        y: 0,
      },
      inputPorts: [{ id: "node-1-input-1" }],
      outputPorts: [{ id: "node-1-output-1" }],
    };
    const position = { x: 0, y: 0 };
    const isSelected = false;
    const inputPorts = [{ id: "node-1-input-1" }];
    const outputPorts = [{ id: "node-1-output-1" }];
    const onMouseUp = () => null;
    const onMouseDown = () => null;
    const PortComponent = () => null;

    const tree = renderer
      .create(
        <Node
          node={node}
          position={position}
          isSelected={isSelected}
          inputPorts={inputPorts}
          outputPorts={outputPorts}
          onMouseUp={onMouseUp}
          onMouseDown={onMouseDown}
          PortComponent={PortComponent}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
