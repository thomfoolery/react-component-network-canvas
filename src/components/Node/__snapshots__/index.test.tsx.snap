// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node renders correctly 1`] = `
<div
  className="Node"
  id="Node-node-1"
  style={
    Object {
      "left": 0,
      "top": 0,
    }
  }
>
  <div
    className="NodeInputPorts"
  />
  <div
    className="NodeBody"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  />
  <div
    className="NodeOutputPorts"
  />
</div>
`;
